root/programs/C#/Frameworks/Infrastructure/Framework/Transmission/CallController.cs

```diff
@@ -1253,7 +847,233 @@ public object Invoke(string serviceName, object parameterValue)
+            {
+                // ユーザ指定：なし
+
+                if (!props.ContainsKey(FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_URL))// Dic化でnullチェック変更
+                {
+                    // XML定義：キーが無い
+                }
+                else
+                {
+                    if (props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_URL] == null
+                        || (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_URL] == "")
+                    {
+                        // XML定義：null or 空文字列
+                    }
+                    else
+                    {
+                        // XML定義：あり
+
+                        // プロキシを生成（XML定義）
+                        proxy = new WebProxy(new Uri((string)props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_URL]));
+                    }
+                }
+            }
+            else
+            {
+                // ユーザ指定：あり
+
+                // プロキシを生成（ユーザ指定）
+                proxy = new WebProxy(new Uri(this._proxyUrl));
+            }
+
+            #endregion
+
+            #region プロキシのオプション設定
+
+            if (proxy == null)
+            {
+                // プロキシがない
+            }
+            else
+            {
+                // プロキシがある
+
+                // ローカル プロキシをバイパスの有効・無効（Default：false）。 
+                proxy.BypassProxyOnLocal = false; // 無効
+
+                // 実行アカウントでのWindows認証の有効・無効（Default：false）。                    
+                proxy.UseDefaultCredentials = true; // 有効
+
+                #region Proxyのクライアント認証のセキュリティ資格情報
+
+                // Proxyのセキュリティ資格情報
+                if (this._nwcProxy == null)
+                {
+                    // ユーザ指定：なし
+
+                    if (!props.ContainsKey(FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_USER_NAME))// Dic化でnullチェック変更
+                    {
+                        // XML定義：キーが無い
+                    }
+                    else
+                    {
+                        if (props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_USER_NAME] == null
+                            || (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_USER_NAME] == "")
+                        {
+                            // XML定義：null or 空文字列
+                        }
+                        else
+                        {
+                            // XML定義：あり
+
+                            // Proxyのセキュリティ資格情報を生成する。
+                            NetworkCredential nwcProxy = new NetworkCredential();
+
+                            nwcProxy.UserName = (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_USER_NAME];
+                            nwcProxy.Password = (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_PASSWORD];
+
+                            // 省略可能に変更した。
+                            if (props.ContainsKey(FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_DOMAIN))
+                            {
+                                nwcProxy.Domain = (string)props[FxLiteral.TRANSMISSION_HTTP_PROP_PROXY_DOMAIN];
```

Comment:
@daisukenishino 

Similar to [this](https://github.com/OpenTouryoProject/OpenTouryo/pull/253/files#r134915287).